tests/v1/e2e/test_async_scheduling.py

```diff
@@ -231,6 +278,15 @@ def run_test(
     print("-" * 80)
     print(f"---- TESTING {test_str}: {test_config}")
     print("-" * 80)
+
+    # On ROCm: use float16 for first test (ROCM_AITER_FA), but float32 for
+    # spec decoding test (TRITON_ATTN) for better precision.
+    # On others: always use float32.
+    if current_platform.is_rocm() and not is_testing_with_spec_decoding:
+        dtype = "float16"
+    else:
+        dtype = "float32"
```

Comment:
Why not use float32 for everything? Would you still have to update tolerances for rocm if everything was float32?

Reply:
I think that only FLEX_ATTENTION works for NVIDIA on this test. That's the only attention backend that is more numerically accurate than others. So all other backends don't work that well. Other than that, our backends are tested on float16, bfloat16, fp8, and in some cases fp4. So for now and for ROCm we are going to go with the float16 precision. In the future we will introduce a more accurate backend to better serve fp32 workloads.